pulp_rpm/app/metadata_parsing.py

```diff
@@ -123,3 +126,160 @@ def process_other_package_element(element):
             changelogs.append((author, date, text))
 
     return pkgid, changelogs
+
+
+def warningcb(warning_type, message):
+    """Optional callback for warnings about wierd stuff and formatting in XML.
+
+    Args:
+        warning_type (int): One of the XML_WARNING_* constants.
+        message (str): Message.
+    """
+    log.warn("PARSER WARNING: %s" % message)
+    return True  # continue parsing
+
+
+def parse_repodata(primary_xml_path, filelists_xml_path, other_xml_path, only_primary=False):
+    """
+    Parse repodata to extract package info.
+
+    Args:
+        primary_xml_path(str): a path to a downloaded primary.xml
+        filelists_xml_path(str): a path to a downloaded filelists.xml
+        other_xml_path(str): a path to a downloaded other.xml
+
+    Returns:
+        dict: createrepo_c package objects with the pkgId as a key
+
+    """
+
+    def pkgcb(pkg):
+        """
+        A callback which is used when a whole package entry in xml is parsed.
+
+        Args:
+            pkg(preaterepo_c.Package): a parsed metadata for a package
+
+        """
+        packages[pkg.pkgId] = pkg
+
+    def newpkgcb(pkgId, name, arch):
+        """
+        A callback which is used when a new package entry is encountered.
+
+        Only opening <package> element is parsed at that moment.
+        This function has to return a package which parsed data will be added to
+        or None if a package should be skipped.
+
+        pkgId, name and arch of a package can be used to skip further parsing. Available
+        only for filelists.xml and other.xml.
+
+        Args:
+            pkgId(str): pkgId of a package
+            name(str): name of a package
+            arch(str): arch of a package
+
+        Returns:
+            createrepo_c.Package: a package which parsed data should be added to.
+
+            If None is returned, further parsing of a package will be skipped.
+
+        """
+        return packages.get(pkgId, None)
+
+    packages = collections.OrderedDict()
+
+    cr.xml_parse_primary(primary_xml_path, pkgcb=pkgcb, warningcb=warningcb, do_files=False)
+    if not only_primary:
+        cr.xml_parse_filelists(filelists_xml_path, newpkgcb=newpkgcb, warningcb=warningcb)
+        cr.xml_parse_other(other_xml_path, newpkgcb=newpkgcb, warningcb=warningcb)
+    return packages
+
+
+class MetadataParser:
+    """Parser for RPM metadata."""
+
+    def __init__(self):
+        """Initialize empty (use one of the alternate constructors)."""
+        self.primary_xml_path = None
+        self.filelists_xml_path = None
+        self.other_xml_path = None
+
+    @staticmethod
+    def from_metadata_files(primary_xml, filelists_xml, other_xml):
```

Comment:
Looking at the code below, it seems that all parameters are paths.
We call the them `*_path` in other places, does it make sense to do it here as well?
Alternatively, a doc string can clarify this.